src/Ampersand/Prototype/GenAngularFrontend.hs

```diff
@@ -1,33 +1,243 @@
+
+            return . T.intercalate eol
+              . concatMap indentEOL -- flatten 2d array
+              . T.lines
+              . renderTemplate (Just . btKeys $ header) parentTemplate
+              $ atomicAndBoxAttrs
+                . setAttribute "isRoot" (depth == 0)
+                . setAttribute "subObjects" subObjAttrs
+    FEObjT {} -> pure ""
+  where
+    genView_SubObject :: (HasRunner env, HasDirPrototype env) => FEObject -> RIO env SubObjectAttr2
+    genView_SubObject subObj =
+      case subObj of
+        FEObjE {} ->
+          do
+            lns <- genViewObject fSpec (depth + 1) subObj
+            return
+              SubObjAttr
+                { subObjName = escapeIdentifier $ objName subObj,
+                  subObjLabel = objName subObj, -- no escaping for labels in templates needed
+                  subObjContents = lns,
+                  subObjExprIsUni = exprIsUni subObj
+                }
+        FEObjT {} ->
+          do
+            return
+              SubObjAttr
+                { subObjName = escapeIdentifier $ objName subObj,
+                  subObjLabel = objName subObj,
+                  subObjContents = objTxt subObj,
+                  subObjExprIsUni = True
+                }
+
+    getTemplateForObject ::
+      (HasDirPrototype env) =>
+      RIO env FilePath
+    getTemplateForObject
+      | relIsProp obj && (not . exprIsIdent) obj -- special 'checkbox-like' template for propery relations
+        =
+        return $ "View-PROPERTY" <> ".html"
+      | otherwise = getTemplateForConcept . target . objExp $ obj
+    getTemplateForConcept ::
+      (HasDirPrototype env) =>
+      A_Concept ->
+      RIO env FilePath
+    getTemplateForConcept cpt = do
+      exists <- doesTemplateExist cptfn
+      return $
+        if exists
+          then cptfn
+          else "Atomic-" <> show ttp <.> "html"
+      where
+        ttp = cptTType fSpec cpt
+        cptfn = "Concept-" <> T.unpack (name cpt) <.> "html"
+
+-- This function is a helper function to add indentation using the EOL character sequence
+-- Let us explain why! For the html view generator we are using HStringTemplates recursively;
+-- other templates are used inside the 'blanks' of parent templates. The subtemplates need to be
+-- indentated properly. However, the level of indentation (number of spaces) is determined by
+-- the position in its parent templates, all the way up to the outmost template.
+-- The solution we came up with:
+--   * After rendering each (sub)template we replace the newlines (\n) by a EOL character sequence, resulting in a single line
+--   * When this result is rendered in a parent template, we end up with a multi-line text with the sub results still on single lines
+--   * For each line in the text, we post process the line, splitting based on EOL character and prefixing the lines (except the first)
+--   * The resulting text is indented correctly
+indentEOL :: Text -> [Text]
```

Comment:
There are several problems with this function `indentEOL`.

1. This is based on a magic string, `<<EOL>>`. In the unlikely, but possible case that a template contains such a text, it will break the code. 
2. You have to be very cautions if you use functions from Partial. In this case, there is no justification for the use of head and tail. With this implementation, indentEOL will crash at runtime if an empty template is provided. Rewrite the function without using head and tail.  The use of splitOn can be justified because you know that `eol` is not empty.

Reply:
1. In that indeed unlikely situation the code will not break, we'll have additional indentation that is wrong, but acceptable. Let's leave it as is.
2. Head and tail replaced by @hanjoosten.